### PR TITLE
Build libgui

### PR DESCRIPTION
**Pull request by vicamo**

### CODE REVIEW

**Comment by vicamo** [comment]
Unresolved symbols:
```
dpkg-shlibdeps: warning: symbol eglWaitSyncKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglDestroyImageKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglCreateImageKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglClientWaitSyncKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglCreateSyncKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglDestroySyncKHR used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol eglDupNativeFenceFDANDROID used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol _Z35eglQueryStringImplementationANDROIDPvi used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
dpkg-shlibdeps: warning: symbol glEGLImageTargetTexture2DOES used by debian/libandroid-gui0/usr/lib/x86_64-linux-gnu/libandroid-gui.so.0.2.0 found in none of the libraries
```

**Comment by vicamo** [comment]
Except from **eglDupNativeFenceFDANDROID**, **Z35eglQueryStringImplementationANDROIDPvi** and **glEGLImageTargetTexture2DOES**, all others can be accessed with **eglGetProcAddress**.

**Comment by vicamo** [comment]
**eglQueryStringImplementationANDROID** should be replaced by **eglQueryString** I think.

**Comment by vicamo** [comment]
For **eglDupNativeFenceFDANDROID**, it seems it's always correctly guarded by **eglCreateSyncKHR**. So until we have implemented native fence feature for the android platform in MESA, it's safe to assume that function will be available through **eglGetProcAddress**.

As for **glEGLImageTargetTexture2DOES**, there are several reference code snippets that use **eglGetProcAddress** to retrieve function pointer of it, too.

So basically I have all the solutions for the unresolved references here. Except I don't really know if they work ;)

**Comment by vicamo** [comment]
So Android libEGL exports all the aforementioned, egl-prefixed functions, and libGLESv1_CM/libGLESv2 exports glEGLImageTargetTexture2DOES, but MESA doesn't. I could create an small shared library named libandroid-EGL and so on to handle such symbols, but it seems to be an overkill?
